src/Temporalio/Bridge/src/worker.rs

```diff
@@ -503,6 +509,26 @@ impl TryFrom<&WorkerOptions> for temporal_sdk_core::WorkerConfig {
             .max_concurrent_wft_polls(opt.max_concurrent_workflow_task_polls as usize)
             .nonsticky_to_sticky_poll_ratio(opt.nonsticky_to_sticky_poll_ratio)
             .max_concurrent_at_polls(opt.max_concurrent_activity_task_polls as usize)
+            .workflow_failure_errors(if opt.nondeterminism_as_workflow_fail {
+                HashSet::from([WorkflowErrorType::Nondeterminism])
+            } else {
+                HashSet::new()
+            })
+            .workflow_types_to_failure_errors(
+                opt.nondeterminism_as_workflow_fail_for_types
+                    .to_option_str()
+                    .map(|s| {
+                        s.split('\n')
```

Comment:
You could make the to map helper for this generic and use it here.

I'm still not a fan of parsing on newlines like this, but it's not hugely important to change.

Reply:
I already have a map helper for metadata (and it's newline delimited too), but I admit being lazy, because getting collections of strings right in a non-annoying way over FFI boundary requires basically building your own structures (or saying something like `\0` delimited) and I was hoping to avoid it.

Reply:
I just added string arrays in [this commit](https://github.com/cretz/temporal-sdk-dotnet/commit/07524da4cfb568d61213f58420b324c5b1d10a31) and removed the newline restriction